src/main/java/io/cdap/plugin/aws/sqs/source/SqsSourceConfig.java

```diff
@@ -0,0 +1,440 @@
+          .withConfigProperty(PROPERTY_ACCESS_KEY);
+      }
+    }
+
+    validateSqsConnection(collector);
+  }
+
+  @VisibleForTesting
+  void validateSqsConnection(FailureCollector collector) {
+    try {
+      AmazonSQS sqs = SqsUtil.getSqsClient(getAuthenticationMethod(), accessId, accessKey, sqsEndpoint, region,
+        queueName);
+      if (sqs != null) {
+        sqs.shutdown();
+      }
+    } catch (SqsInitializationException e) {
+      collector.addFailure("Unable to connect to Amazon SQS.",
+        "Ensure properties like Auth Method, Access ID, Access Key, Region, Queue name are correct.")
+        .withConfigProperty(PROPERTY_ACCESS_ID)
+        .withConfigProperty(PROPERTY_ACCESS_KEY)
+        .withConfigProperty(PROPERTY_REGION)
+        .withConfigProperty(PROPERTY_QUEUE_NAME)
+        .withStacktrace(e.getStackTrace());
+    }
+  }
+
+  private void validateQueueName(FailureCollector collector) {
+    if (containsMacro(PROPERTY_QUEUE_NAME)) {
+      return;
+    }
+
+    if (Utility.isNullOrEmpty(queueName)) {
+      collector.addFailure("Queue name must be specified.", null)
+        .withConfigProperty(PROPERTY_QUEUE_NAME);
+    }
+  }
+
+  private void validateRegion(FailureCollector collector) {
+    if (containsMacro(PROPERTY_REGION)) {
+      return;
+    }
+
+    getRegion(collector);
+  }
+
+  private void validateInterval(FailureCollector collector) {
+    if (containsMacro(PROPERTY_INTERVAL)) {
+      return;
+    }
+
+    if (interval < 0) {
+      collector.addFailure(String.format("Invalid interval '%d'.", interval),
+        "Ensure the interval is not less than 0")
+        .withConfigProperty(PROPERTY_INTERVAL);
+    }
+  }
+
+  private void validateWaitTime(FailureCollector collector) {
+    if (containsMacro(PROPERTY_WAIT_TIME)) {
+      return;
+    }
+
+    if (waitTime < 1 || waitTime > MAX_WAIT_TIME) {
+      collector.addFailure(String.format("Invalid wait time '%d'.", waitTime),
+        String.format("Ensure the wait time is 1 or at most '%d'", MAX_WAIT_TIME))
+        .withConfigProperty(PROPERTY_WAIT_TIME);
+    }
+  }
+
+  private void validateMaxNumberOfMessages(FailureCollector collector) {
+    if (containsMacro(PROPERTY_NUMBER_OF_MESSAGES)) {
+      return;
+    }
+
+    if (numberOfMessages < 1 || numberOfMessages > MAX_MESSAGES_LIMIT) {
+      collector.addFailure(String.format("Invalid number of messages '%d'.", numberOfMessages),
+        String.format("Ensure the number of messages is 1 or at most '%d'", MAX_MESSAGES_LIMIT))
+        .withConfigProperty(PROPERTY_NUMBER_OF_MESSAGES);
+    }
+  }
+
+  private void validateSchema(FailureCollector collector) {
+    if (containsMacro(PROPERTY_SCHEMA)) {
+      return;
+    }
+
+    if (Utility.isNullOrEmpty(schema)) {
+      collector.addFailure("Schema must be specified.", null)
+        .withConfigProperty(PROPERTY_SCHEMA);
+    }
+  }
+
+  private void validateFormat(FailureCollector collector) {
```

Comment:
Seems like `validateSchema()` can be combined with this function, they seem tightly coupled. Is there a use case for having them separate?

Reply:
Done